common/src/main/java/com/wynntils/core/overlays/objects/OverlayHolder.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.overlays.objects;
+
+import com.wynntils.core.config.objects.StorageHolder;
+import com.wynntils.core.features.properties.FeatureInfo;
+import com.wynntils.core.overlays.properties.Overlay;
+import java.lang.reflect.Field;
+
+public class OverlayHolder extends StorageHolder {
+    private final Overlay metadata;
+
+    public OverlayHolder(Object parent, Field field, Overlay metadata) {
+        super(
+                parent,
+                field,
+                field.getType(),
+                parent.getClass().getAnnotation(FeatureInfo.class).category(),
+                false);
+
+        this.metadata = metadata;
+    }
+
+    public Overlay getMetadata() {
+        return metadata;
+    }
+
+    public boolean getToggled() {
+        // Type casting to allow #isToggled to be called, it should be one anyway
+        return ((OverlayPosition) getValue()).isToggled();
```

Comment:
I have so many questions about this. What is it that should be toggled? Visibility? And why not using standard bean conventions with `is` instead of `get` for booleans? And this casting looks really sus. If you are sure the value is of type OverlayPosition, why not assert it in the constructor? You're just passing on a `field.getType()`.

Reply:
I'm more than happy to replace all `get` to `is`